pkg/controller/plan/adapter/ovirt/client.go

```diff
@@ -281,6 +278,46 @@ func (r *Client) getDiskSnapshot(diskID, targetSnapshotID string) (diskSnapshotI
 	return
 }
 
+func (r *Client) getJobs(correlationID string) (ovirtJob []*ovirtsdk.Job, err error) {
+
+	// TODO: figure out a better way
+	r.connect()
```

Comment:
Why is this connecting here?

Reply:
Otherwise the client is not connected and API calls are NPE'd, I believe it's because the connection is closed outside, in migratio.go

Reply:
Did you test this with more than one VM on the migration plan? This is not safe.

A single Client is instantiated at the start of `Migration.Run()` (in `Migration.init()`), which is then closed in a defer at the end of `Migration.Run()`. Inside of `Run()`, `Migration.execute()` gets called for each VM migration on the plan that is currently in flight, plus once more if there's an additional migration ready and able to start. If you had only a single VM on the migration plan and the remove snapshot goroutine was long running, it's likely that `Run()` would return and the connection would be closed before the goroutine finished. However, it's possible for there to be many VM migrations running, and all of them use the same client in a given reconcile. These VM migrations can all be at different steps in the process, so it's entirely possible with this PR that your first VM could finish, delete its snapshots, and then call `client.Close()` inside of `removePrecopies()`, which would mean that the client was closed for all of the following calls to `execute()`.

You need to be sure that the goroutine isn't sharing the client with anything else. Since you're kicking off this goroutine from inside of the client, you'll probably need to rethink this somewhat.

Reply:
Thanks Sam, and yes, it is a problem I was aware of, I just wasn't sure how to approach it before and decided to go on with the actual snapshot removal first

anyway, I've updated this PR and made it into a slightly larger change. It feels like the main issue is that because of oVirt's shenanigans using RemoveSnapshots is a bit too high of a denominator because of the need for more control for how it is executed.

So what I went with is to add a new interface method to finalize the plan in (`end()`) where oVirt's client will start the asynchronous sequential snapshot removal and can managed the lifecycle of the connection to the SDK. Now, RemoveSnapshots() will do nothing for oVirt.

I've tested it with multiple VMs and it looks ok, I'm not sure about whether this is the best way to do this or if I placed it correctly, but this is what we came up with trying to both control the lifetime of the connection and not have a new connection created for each VM